### PR TITLE
networkminer: init at 2.8

### LINKED ISSUE #233414: Package request: NetworkMiner
**NetworkMiner**

NetworkMiner is an [open source](https://www.netresec.com/?page=NetworkMinerSourceCode) network forensics tool that extracts artifacts, such as files, images, emails and passwords, from captured network traffic in PCAP files. NetworkMiner can also be used to capture live network traffic by sniffing a network interface. Detailed information about each IP address in the analyzed network traffic is aggregated to a network host inventory, which can be used for passive asset discovery as well as to get an overview of which devices that are communicating. NetworkMiner is primarily designed to run in Windows, but can also be used in [Linux](https://www.netresec.com/?page=Blog&month=2014-02&post=HowTo-install-NetworkMiner-in-Ubuntu-Fedora-and-Arch-Linux).

**Metadata**

* homepage URL: https://www.netresec.com/?page=NetworkMiner
* source URL: https://www.netresec.com/?page=Blog&month=2014-02&post=HowTo-install-NetworkMiner-in-Ubuntu-Fedora-and-Arch-Linux
* platforms: linux

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

The Open Source Network Forensic Analysis Tool (NFAT) https://www.netresec.com/?page=NetworkMiner

Closes #233414

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emilytrau** [comment]
@KCynk would you please test let me know if you run into any issues :)

File: pkgs/tools/security/networkminer/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  nugetDeps = ./deps.nix;
+
+  buildPhase = ''
+    runHook preBuild
+
+    msbuild /p:Configuration=Release NetworkMiner.sln
+
+    runHook postBuild
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/security/networkminer/default.nix:45]: We can't use the standard options for this?

File: pkgs/tools/security/networkminer/default.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    # Upstream does not provide versioned releases, a mirror has been uploaded
+    # to archive.org
+    url = "https://archive.org/download/networkminer-${builtins.replaceStrings ["."] ["-"] version}/NetworkMiner_${builtins.replaceStrings ["."] ["-"] version}_source.zip";
+    sha256 = "1n2312acq5rq0jizlcfk0crslx3wgcsd836p47nk3pnapzw0cqvv";
+  };
+
+  nativeBuildInputs = [ unzip dos2unix msbuild ];
+
+  patches = [
+    # Store application data in XDG_DATA_DIRS instead of trying to write to nix store
```
Comment (SuperSandro2000) [inline pkgs/tools/security/networkminer/default.nix:24]: Can we try to upstream this?

File: pkgs/tools/security/networkminer/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+, msbuild
+, mono
+}:
+buildDotnetModule rec {
+  pname = "networkminer";
+  version = "2.8";
+
+  src = fetchurl {
+    # Upstream does not provide versioned releases, a mirror has been uploaded
+    # to archive.org
```
[the remark this answers is not in the document]
Comment (emilytrau) [reply pkgs/tools/security/networkminer/default.nix:16]: netresec only serves downloads to browser user agents, which web.archive.org can't snapshot :(

File: pkgs/tools/security/networkminer/default.nix (hunk 2 above)
Comment (emilytrau) [reply pkgs/tools/security/networkminer/default.nix:24]: The upstream project does not appear to accept contributions...

File: pkgs/tools/security/networkminer/default.nix (hunk 1 above)
Comment (emilytrau) [reply pkgs/tools/security/networkminer/default.nix:45]: I could only get this to work with `msbuild` and not with the standard `dotnet build`. Maybe something to do with using mono?

**Review by mdarocha — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**